include/envoy/network/listen_socket.h

```diff
@@ -15,9 +16,18 @@
 namespace Envoy {
 namespace Network {
 
-// Optional variant of setsockopt(2) optname. The idea here is that if the option is not supported
-// on a platform, we can make this the empty value. This allows us to avoid proliferation of #ifdef.
-using SocketOptionName = absl::optional<std::pair<int, int>>;
+// SocketOptionName is an optional value that captures the setsockopt(2)
+// arguments. The idea here is that if a socket option is not supported
+// on a platform, we can make this the empty value. This allows us to
+// avoid proliferation of #ifdef. The first element is the sockopt
+// level, the second is the optname and the last is the human-readable
+// name for error logs.
+using SocketOptionName = absl::optional<std::tuple<int, int, std::string>>;
```

Comment:
Mild preference for using a struct here vs. a tuple since IMO it's easier to read.

Reply:
Yeh I went tuple because it began with pair, but I agree a struct is better.